### PR TITLE
Add .moveNode() to MutableDocument & DocumentEditorTransaction

### PR DESCRIPTION
**Pull request by roughike**
We want to have a way of moving a `DocumentNode` to an arbitrary index in the list of nodes with just one `notifyListeners()` call happening. 

The end result can be achieved with `document..deleteNode()..insertNode()` calls. But we only want to get a notification of the document after a `DocumentNode` has been moved, not a "partial state" where a node has been removed, and then the "final state" where the node exists on the right place of the document. This can't be achieved currently with the existing APIs.

Since there's no `moveNode()` method available, this PR introduces just that.

### CODE REVIEW

File: super_editor/test/src/core/document_editor_test.dart [hunk 1]
```diff
@@ -3,6 +3,50 @@ import 'package:super_editor/super_editor.dart';
 
 void main() {
   group('MutableDocument', () {
+    test('it moves node from its current index to the given index ', () {
```
Comment (matthew-carroll) [inline super_editor/test/src/core/document_editor_test.dart:6]: Can you add a few tests for boundary conditions:
 * what happens when this is called on an empty document
 * what happens when this is called with an invalid node ID
 * what happens when the node is moved to a negative index
 * what happens when the node is moved to an index beyond the length of the document

File: super_editor/lib/src/core/document_editor.dart [hunk 2]
```diff
@@ -273,6 +279,15 @@ class MutableDocument with ChangeNotifier implements Document {
     return isRemoved;
   }
 
+  /// Moves the given [node] from its current index in the [Document] to the
+  /// given [toIndex].
+  void moveNode({required DocumentNode node, required int toIndex}) {
+    if (nodes.remove(node)) {
+      nodes.insert(toIndex, node);
+      notifyListeners();
+    }
```
[the remark this answers is not in the document]
Comment (roughike) [reply super_editor/lib/src/core/document_editor.dart]: I went with the nodeId & targetIndex approach. I'm assuming that the `nodeId` of the node you want to move is always known to you, but figuring out the index of the node might usually be an extra step.

File: super_editor/test/src/core/document_editor_test.dart (hunk 1 above)
Comment (roughike) [reply super_editor/test/src/core/document_editor_test.dart]: I wrote the missing tests.

**Review by matthew-carroll — APPROVED** [review]
LGTM
